policy-engine/src/main.rs

```diff
@@ -45,18 +52,27 @@ fn main() -> Result<(), Error> {
                 2 => LevelFilter::Debug,
                 _ => LevelFilter::Trace,
             },
-        )
-        .init();
+        ).init();
+
+    // Metrics service.
+    server::new(move || {
+        App::new()
+            .middleware(Logger::default())
+            .route("/metrics", Method::GET, metrics::serve)
+    }).bind((opts.metrics_address, opts.metrics_port))?
+    .start();
 
+    // Main service.
     let state = graph::State {
         upstream: opts.upstream,
     };
     server::new(move || {
         App::with_state(state.clone())
             .middleware(Logger::default())
             .route("/v1/graph", Method::GET, graph::index)
-    })
-    .bind((opts.address, opts.port))?
-    .run();
+    }).bind((opts.address, opts.port))?
+    .start();
+
+    sys.run();
```

Comment:
Does this end up waiting for both servers?

Reply:
Yes, this runs and waits for both servers (if one is panicking the other keep going).